warehouse/schema/schema.go

```diff
@@ -45,7 +45,21 @@ type fetchSchemaRepo interface {
 	FetchSchema(ctx context.Context) (model.Schema, error)
 }
 
-type Schema struct {
+type Handler interface {
+	SyncRemoteSchema(ctx context.Context, fetchSchemaRepo fetchSchemaRepo, uploadID int64) (bool, error)
+	IsWarehouseSchemaEmpty() bool
+	GetTableSchemaInWarehouse(tableName string) model.TableSchema
+	GetLocalSchema(ctx context.Context) (model.Schema, error)
+	UpdateLocalSchema(ctx context.Context, updatedSchema model.Schema) error
+	UpdateWarehouseTableSchema(tableName string, tableSchema model.TableSchema)
+	GetColumnsCountInWarehouseSchema(tableName string) int
```

Comment:
@shekhar-rudder 
`GetColumnsCountInWarehouseSchema(tableName string)` Full identifier would be schema.table as same tablename can exist in multiple schemas as well. I did not check this function, can you help share if we are using schema as well when using this func?

Reply:
We don't explicitly use schema in this func. But the warehouse schema which gets fetched at the start is specific to a namespace.
https://github.com/rudderlabs/rudder-server/blob/3c7106ee0e380d49fb46a9a0bda4ecbbcf7054c4/warehouse/integrations/redshift/redshift.go#L1270

https://github.com/rudderlabs/rudder-server/blob/3c7106ee0e380d49fb46a9a0bda4ecbbcf7054c4/warehouse/integrations/redshift/redshift.go#L1223